lib/lte_link_control/lte_lc.c

```diff
@@ -892,58 +862,41 @@ int lte_lc_psm_req(bool enable)
 int lte_lc_psm_get(int *tau, int *active_time)
 {
 	int err;
-	struct at_param_list at_resp_list = {0};
-	char buf[AT_CEREG_RESPONSE_MAX_LEN] = {0};
-	struct lte_lc_psm_cfg psm_cfg;
+	char tau_str[9] = {0};
+	char active_time_str[9] = {0};
 
 	if ((tau == NULL) || (active_time == NULL)) {
 		return -EINVAL;
 	}
 
 	/* Enable network registration status with PSM information */
-	err = at_cmd_write(AT_CEREG_5, NULL, 0, NULL);
+	err = nrf_modem_at_printf(AT_CEREG_5);
 	if (err) {
 		LOG_ERR("Could not set CEREG, error: %d", err);
 		return err;
 	}
 
-	/* Read network registration status */
-	err = at_cmd_write(AT_CEREG_READ, buf, sizeof(buf), NULL);
-	if (err) {
-		LOG_ERR("Could not get CEREG response, error: %d", err);
-		return err;
-	}
-
-	err = at_params_list_init(&at_resp_list, AT_CEREG_PARAMS_COUNT_MAX);
-	if (err) {
-		LOG_ERR("Could not init AT params list, error: %d", err);
-		return err;
-	}
-
-	err = at_parser_max_params_from_str(buf,
-					    NULL,
-					    &at_resp_list,
-					    AT_CEREG_PARAMS_COUNT_MAX);
-	if (err) {
-		LOG_ERR("Could not parse AT+CEREG response, error: %d", err);
-		goto parse_psm_clean_exit;
+	/* It's required that both active time and TAU are present to proceed
+	 * with parsing of the PSM values.
+	 */
+	err = nrf_modem_at_scanf(AT_CEREG_READ,
+				 "+CEREG: %*u,%*u,%*u,%*u,%*u,%*u,%*u,\"%8[0-1]\",\"%8[0-1]\"",
+				 active_time_str, tau_str);
```

Comment:
Does this work if there are empty fields in between? `*` is an "assignment-suppressing character", so does the input still have to match the specifier? Also if active time and TAU match, is the returned value really 2 or are also fields marked with `*` counted as matches?

Reply:
Doesn't seem to work, here's what happens with modem_shell:
```
mosh:~$ at at+cereg?
+CEREG: 5,1,"5A00","00023906",7,,,"11100000","11100000"
OK
mosh:~$ link psm --read
Cannot get PSM configs: -14
[00:02:03.884,674] <err> lte_lc: Failed to get +CEREG response, error: -77
```

Reply:
Only the matches without `*` should be counted. Regarding "missing" parameters  I'll need to look more into that. It seems that can cause issues.

Reply:
I have worked around this by using `AT%XMONITOR` instead in the case of reading PSM configuration. The same command is also used now in case of +CEREG notification. This is done to also get the T3412 legacy timer, which is not included in +CEREG.